lib/core.js

```diff
@@ -307,6 +307,15 @@ interface Iterable<T> {
     @@iterator(): Iterator<T>;
 }
 
+/* Generators */
+interface Generator<Y,R,N> {
+    @@iterator(): Iterator<Y>;
+    next(value?: N): IteratorResult<Y|R>;
+}
+
```

Comment:
Super rough idea, but I wonder if we could express this as something like

```JavaScript
interface $Iterator<Y, R, N> {
  next(value?: N): IteratorResult<Y>;
}
type Iterator<T> = $Iterator<T, *, *>

interface $Iterable<Y, R, N> {
  @@iterator(): $Iterator<Y, R, N>;
}
type Iterable<T> = $Iterable<T, *, *>;

// Generator<Y, R, N> should be an $Iterable<Y, R, N>
// Generator<Y, R, N> should be an Iterable<Y>
// Generator<Y, R, N> should be an $Iterator<Y, R, N>
// Generator<Y, R, N> should be an Iterator<Y>
interface Generator<Y, R, N> {
    @@iterator(): Iterator<Y>;
    next(value?: N): IteratorResult<Y|R>;
}

declare function $yieldDelegate<R, N>(g: $Iterable<*, R, N>, n:N): R
```

If we do this, I think we could deal with something like

```JavaScript
yield *({ [Symbol.iterator]: function *() { return yield 4; } })
```

ignoring the fact that Flow doesn't have support for `Symbol.iterator`, of course :P